components/tikv_util/src/mpsc/batch.rs

```diff
@@ -294,7 +314,12 @@ impl<T, E, I: Fn() -> E, C: Fn(&mut E, T)> Stream for BatchReceiver<T, E, I, C>
         } else if self.elem.is_none() {
             return Ok(Async::NotReady);
         }
-        Ok(Async::Ready(self.elem.take()))
+        let elem = self.elem.take();
+        if let Some(m) = received {
+            let collection = self.elem.get_or_insert_with(&self.initializer);
+            assert!(self.collector.collect(collection, m).is_none());
```

Comment:
So the `received` will be included in the next batch?

Reply:
yes.